drivers/storage/azure/utils/utils.go

```diff
@@ -0,0 +1,77 @@
+// +build !libstorage_storage_driver libstorage_storage_driver_azure
+
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"os"
+
+	"github.com/akutz/goof"
+
+	"github.com/codedellemc/libstorage/api/types"
+	"github.com/codedellemc/libstorage/drivers/storage/azure"
+)
+
+func checkAzureMarkInFile(ctx types.Context) bool {
+	file := "/var/lib/dhcp/dhclient.eth0.leases"
+	pattern := []byte("unknown-245")
+
+	f, err := os.Open(file)
+	if err != nil {
+		ctx.Debug("Specific file (" + file + ") could not be opened:")
+		ctx.Debug(err)
+		return false
+	}
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		if bytes.Contains(scanner.Bytes(), pattern) {
+			return true
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		ctx.Debug("Specific file (/var/lib/dhcp/dhclient.eth0.leases) could not be read:")
+		ctx.Debug(err)
+	}
+	return false
+}
+
+// IsAzureInstance returns a flag indicating whether the executing host is an Azure
+// instance based on whether or not the metadata URL can be accessed.
```

Comment:
The comment here does not match how the function is actually implemented. It does not use metadata that I can see.